### PR TITLE
Add aria labels to 526 form locations missing them

### PR DESCRIPTION
**Pull request by ndsprinkle**
## Description
Some of the UI controls for editing and removing sections of content don't have aria-labels. This leaves screen reader users to find headings or other related text when determining what action they are taking. This adds aria labels to the specified locations in the connected issue.

## Definition of done
- [ ] Specified locations are properly labeled with aria labels

### CODE REVIEW

**Comment by ndsprinkle** [comment]
@Mottie This PR takes care of all locations mentioned in the connected issue. Do you think I should have added aria labels to literally all locations missing them or just the ones mentioned in the issue?

**Review by Mottie — CHANGES_REQUESTED** [review]
Yeah, I think it would be a good idea update all the buttons in these files.

File: src/platform/forms-system/src/js/fields/ArrayField.jsx [hunk 1]
```diff
@@ -320,6 +320,7 @@ export default class ArrayField extends React.Component {
                               className="usa-button-secondary float-right"
```
Comment (Mottie) [inline src/platform/forms-system/src/js/fields/ArrayField.jsx]: There is an "Update" button on line 310 that should be updated.

File: src/platform/forms-system/src/js/fields/FileField.jsx [hunk 2]
```diff
@@ -246,6 +246,7 @@ export default class FileField extends React.Component {
                 id={`${idSchema.$id}_add_label`}
                 htmlFor={idSchema.$id}
                 className="usa-button usa-button-secondary"
+                aria-label={uiSchema['ui:title'] || schema.title}
               >
                 <span
```
Comment (Mottie) [inline src/platform/forms-system/src/js/fields/FileField.jsx]: I'm not sure why a `<span>` is being used as a button here. Should this be updated @1Copenut?

Either way, I think it needs an `aria-label` as well.

File: src/platform/forms-system/src/js/fields/FileField.jsx [hunk 3]
```diff
@@ -246,6 +246,7 @@ export default class FileField extends React.Component {
                 id={`${idSchema.$id}_add_label`}
```
Comment (Mottie) [inline src/platform/forms-system/src/js/fields/FileField.jsx]: There is a "Cancel" (upload) button at line 159 that should be updated.

File: src/platform/forms-system/src/js/fields/FileField.jsx (hunk 2 above)
Comment (1Copenut) [reply src/platform/forms-system/src/js/fields/FileField.jsx]: @Mottie We deliberately put the `<span>` in to accept the `role="button"`. Our label covers up the `input[type="file"]` so we needed to ensure the visual "button" announced itself and responded as a button. Adding the role smoothed it out for screen readers, but we had to add it to the span b/c labels can't have their implicit role removed.

We used this article as our guide: https://accessabilly.com/a-styled-accessible-file-upload/

**Review by 1Copenut — CHANGES_REQUESTED** [review]
I'm seeing three more instances where we need aria-labels. On the review page, we have the ability to add facilities, and we have some instances of the label being `Undefined`. Screen shots attached.

---
Screen Shot 2019-10-23 at 2 01 30 PM

---
Screen Shot 2019-10-23 at 2 03 59 PM

---
Screen Shot 2019-10-23 at 2 32 37 PM

**Review by Mottie — APPROVED** [review]

**Review by rianfowler — APPROVED** [review]

File: src/platform/forms-system/src/js/fields/ArrayField.jsx [hunk 4]
```diff
@@ -320,6 +321,7 @@ export default class ArrayField extends React.Component {
...
                               type="button"
                               onClick={() => this.handleRemove(index)}
+                              aria-label={`Remove ${title}`}
```
Comment (1Copenut) [inline src/platform/forms-system/src/js/fields/ArrayField.jsx]: @Mottie I'm going to reverse my field on adding `aria-label` to these Remove buttons. When I listened to them on localhost, it was more confusing to have the longer label in some places and not in others. Let's make all the Remove buttons just say Remove as before.

Apologies for asking for a second round of changes. No further review needed from me after this.

File: src/platform/forms-system/src/js/review/ArrayField.jsx [hunk 5]
```diff
@@ -270,6 +270,7 @@ class ArrayField extends React.Component {
                                 type="button"
                                 className="usa-button-secondary float-right"
                                 onClick={() => this.handleRemove(index)}
+                                aria-label={`Remove ${uiOptions.itemName}`}
```
Comment (1Copenut) [inline src/platform/forms-system/src/js/review/ArrayField.jsx]: Please remove this `aria-label` as well.

**Review by 1Copenut — APPROVED** [review]
I've got two small requests to remove the `aria-label` from our "Remove" buttons. It was more confusing to hear the longer labels in context. No further changes needed. Approved by me after these are restored to previous.

**Comment by ndsprinkle** [comment]
@1Copenut I have removed the aria-labels from the Remove buttons as requested :)

**Comment by ndsprinkle** [comment]
@rianfowler I have made another commit that removes a two aria-labels that I had added. I'm sure it's small enough that you don't need to re-review, but I did want to bring it to your attention.

**Pull request merged**